rust/src/common/error.rs

```diff
@@ -28,34 +28,44 @@ use super::{address::Address, RequestID};
     ConnectionIsClosed() =
-        1: "The connection has been closed and no further operation is allowed.",
+        2: "The connection has been closed and no further operation is allowed.",
     SessionIsClosed() =
-        2: "The session is closed and no further operation is allowed.",
+        3: "The session is closed and no further operation is allowed.",
     TransactionIsClosed() =
-        3: "The transaction is closed and no further operation is allowed.",
+        4: "The transaction is closed and no further operation is allowed.",
     TransactionIsClosedWithErrors(String) =
-        4: "The transaction is closed because of the error(s):\n{}",
+        5: "The transaction is closed because of the error(s):\n{}",
     UnableToConnect() =
-        5: "Unable to connect to TypeDB server.",
+        6: "Unable to connect to TypeDB server.",
     DatabaseDoesNotExist(String) =
-        8: "The database '{}' does not exist.",
+        9: "The database '{}' does not exist.",
     MissingResponseField(&'static str) =
-        9: "Missing field in message received from server: '{}'.",
+        10: "Missing field in message received from server: '{}'.",
     UnknownRequestId(RequestID) =
-        10: "Received a response with unknown request id '{}'",
+        11: "Received a response with unknown request id '{}'",
     InvalidResponseField(&'static str) =
-        11: "Invalid field in message received from server: '{}'.",
+        12: "Invalid field in message received from server: '{}'.",
     ClusterUnableToConnect(String) =
-        12: "Unable to connect to TypeDB Cluster. Attempted connecting to the cluster members, but none are available: '{}'.",
+        13: "Unable to connect to TypeDB Cluster. Attempted connecting to the cluster members, but none are available: '{}'.",
     ClusterReplicaNotPrimary() =
-        13: "The replica is not the primary replica.",
+        14: "The replica is not the primary replica.",
     ClusterAllNodesFailed(String) =
-        14: "Attempted connecting to all cluster members, but the following errors occurred: \n{}.",
+        15: "Attempted connecting to all cluster members, but the following errors occurred: \n{}.",
     ClusterTokenCredentialInvalid() =
-        16: "Invalid token credential.",
+        17: "Invalid token credential.",
     SessionCloseFailed() =
-        17: "Failed to close session. It may still be open on the server: or it may already have been closed previously.",
+        18: "Failed to close session. It may still be open on the server: or it may already have been closed previously.",
+    ClusterEndpointEncrypted() =
+        19: "Unable to connect to TypeDB Cluster: attempting an unencrypted connection to an encrypted endpoint.",
+    ClusterSSLCertificateNotValidated() =
+        20: "SSL handshake with TypeDB Cluster failed: the server's identity could not be verified.",
```

Comment:
Can we write something easier to action in this message? Does it mean that the root CA does not match, or something else?

Reply:
> Does it mean that the root CA does not match

I believe so. How about "SSL handshake with TypeDB Cluster failed: the server's identity could not be verified. Possible CA mismatch."?

Reply:
Sure, that sounds good